moonstone/filtering/mean_filtering.py

```diff
@@ -0,0 +1,120 @@
+        logger.info('Started with %i items and a total of %i reads' % (self.raw_items_number, self.raw_reads_number))
+        logger.info('Saving new matrix with %i items and %i reads' % (self.remaining_items_number,
+                                                                      self.remaining_reads_number))
+        logger.info('Retained %.2f%% of items and %.2f%% of reads' % (
+            self.remaining_items_number / self.raw_items_number * 100,
+            self.remaining_reads_number / self.raw_reads_number * 100)
+            )
+        return filtered_df
+
+    def plot_threshold_vs_remaining_data(self, write_html_file, items_name='items'):
+        '''  The x and y set below are are are either integers or floats.
+        Be aware that some operation will require an np.array  '''
+        x_items = list(self._items_dict.keys())  # get x values for plotting
+        x_reads = list(self._reads_dict.keys())
+        y_items = list(self._items_dict.values())  # get y values for plotting
+        y_reads = list(self._reads_dict.values())
+        '''  The figure visualizing the data and the filtering.  '''
+        fig = make_subplots(specs=[[{"secondary_y": True}]])  # To set a second y-axis.
+        fig.add_trace(
+            go.Scatter(x=x_items, y=y_items, name="Retained Items: left axis"),
+            secondary_y=False,
+        )
+        fig.add_trace(
+            go.Scatter(x=x_reads, y=y_reads, name="Retained Reads: right axis"),
+            secondary_y=True,
+        )
+        fig.add_trace(
+            go.Scatter(x=[self.threshold, self.threshold], y=[0, self.raw_reads_number], name="90% Reads Threshold"),
+            secondary_y=True,
+        )
+        # Add figure title
+        fig.update_layout(
+            title_text="Number of %s and reads in function of the threshold value" % items_name,
+            title_x=0.5
+        )
+        # Set x-axis title
+        fig.update_xaxes(title_text="threshold value")
+        # Set y-axes titles
+        fig.update_yaxes(title_text="number of %s" % items_name, secondary_y=False)
+        fig.update_yaxes(title_text="number of reads", secondary_y=True)
+        fig.show()
+
+        if write_html_file:
+            plotly.io.write_html(fig, write_html_file)
+
+    def visualize(self, write_html_file=False):
+        print(write_html_file)
+        try:
+            self._items_dict
+        except AttributeError:
+            FS_instance = FilteringStats(self.df)
+            self._items_dict, self._reads_dict = FilteringStats.by_mean(FS_instance)
+        self.plot_threshold_vs_remaining_data(write_html_file)
+
+    def generate_report_data(self):
+        data_dic = {
+            "threshold": self.threshold,
+            "n_items_removed": self.raw_items_number-self.remaining_items_number,
+            "n_reads_removed": self.raw_reads_number-self.remaining_reads_number
+            }
+        try:
+            self.percentage_to_keep
```

Comment:
I am not sure this line is necessary? This approach works but try/except should probably more used when dealing with real issues or problems that could occur.

I would suggest you check whether there is a `percentage_to_keep` attribute in your instance such as : 

```python
if getattr(self, 'percentage_to_keep', None) is not None:
    data_dic['percentage_to_keep'] = self.percentage_to_keep
```